src/jit/gcinfo.cpp

```diff
@@ -713,18 +712,13 @@ GCInfo::WriteBarrierForm GCInfo::gcWriteBarrierFormFromTargetAddress(GenTree* tg
         // No need for a GC barrier when writing to a local variable.
         return GCInfo::WBF_NoBarrier;
     }
-    if (tgtAddr->OperGet() == GT_LCL_VAR || tgtAddr->OperGet() == GT_REG_VAR)
+    if (tgtAddr->OperGet() == GT_LCL_VAR)
     {
         unsigned lclNum = 0;
         if (tgtAddr->gtOper == GT_LCL_VAR)
         {
             lclNum = tgtAddr->gtLclVar.gtLclNum;
```

Comment:
Since the case below is now gone, this should have an `assert(tgtAddr->gtOper == GT_LCL_VAR);`

Reply:
Looks like not even an assert is needed since it's obvious that  tgtAddr is GT_LCL_VAR already.

Reply:
Oh, right!

Reply:
Similar to the previous case, you should replace these 5 lines with:
```
unsigned lclNum = tgtAddr->AsLclVar()->GetLclNum();
```